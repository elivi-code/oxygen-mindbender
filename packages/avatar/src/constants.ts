export const avatarSize = {
  size3xsmall: '3xsmall',
  size2xsmall: '2xsmall',
  sizeXsmall: 'xsmall',
  sizeSmall: 'small',
  sizeMedium: 'medium',
  sizeLarge: 'large',
  sizeXlarge: 'xlarge',
  size2xlarge: '2xlarge',
  size3xlarge: '3xlarge',
} as const;

export const avatarUserStatus = {
  available: 'available',
  away: 'away',
  busy: 'busy',
  doNotDisturb: 'doNotDisturb',
  onBreak: 'onBreak',
  onCall: 'onCall',
  onDirectCall: 'onDirectCall',
  offline: 'offline',
  workingOffline: 'workingOffline',
  wrapUp: 'wrapUp',
} as const;
