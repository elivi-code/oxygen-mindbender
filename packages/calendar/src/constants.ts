/** @deprecated use exported type DisplayMode instead */
const DISPLAY_MODE = {
  DATE: 'date',
  DATE_RANGE: 'dateRange',
} as const;

export default DISPLAY_MODE;
