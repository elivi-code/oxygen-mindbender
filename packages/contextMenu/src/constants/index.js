const THEME_NAME = 'contextMenu';
const CONTEXT_MENU_SHOW = 'OXYGEN_CONTEXT_MENU_SHOW';
const CONTEXT_MENU_HIDE = 'OXYGEN_CONTEXT_MENU_HIDE';

export { CONTEXT_MENU_SHOW, CONTEXT_MENU_HIDE, THEME_NAME };
