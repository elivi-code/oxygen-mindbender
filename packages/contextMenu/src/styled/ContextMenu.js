import styled from 'styled-components';

const ContextMenu = styled.div`
  ${({ isDisabled }) => isDisabled && 'cursor: not-allowed;'}
  position: relative;
  display: flex;
  flex-direction: column;
  z-index: ${({ theme }) => theme.contextMenuZIndex};

  width: ${({ theme }) => theme.contextMenuWidth};
  padding: ${({ theme }) => theme.contextMenuPadding};
  border-radius: ${({ theme }) => theme.contextMenuBorderRadius};
  margin: 1px;
  border: ${({ theme }) => theme.contextMenuBorder};

  background-color: ${({ theme }) => theme.contextMenuBackground};
  box-shadow: ${({ theme }) => theme.contextMenuBoxShadow};
`;

export default ContextMenu;
