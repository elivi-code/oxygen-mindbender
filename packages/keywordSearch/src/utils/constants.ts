const SEARCH_RESULT_LIST_ID = 'table-search-results';
const SEARCH_RESULTS_SIZE_LIMIT = 8;

export { SEARCH_RESULTS_SIZE_LIMIT, SEARCH_RESULT_LIST_ID };
