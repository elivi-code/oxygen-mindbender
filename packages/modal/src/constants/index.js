export { default as MODAL_HEADER_ALIGN_TITLE } from './titleAlignment';
export { default as MODAL_SIZE } from './modalSize';
export { default as MODAL_FOOTER_ALIGN } from './footerAlignment';
