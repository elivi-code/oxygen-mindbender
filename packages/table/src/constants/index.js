const THEME_NAME = 'table';

export default THEME_NAME;
