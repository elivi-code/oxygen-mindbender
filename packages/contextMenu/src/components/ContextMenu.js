import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { withTheme, getTestAttributes } from '@8x8/oxygen-config';
import { Manager, Popper } from 'react-popper';
import PubSub from 'pubsub-js';
import noop from 'lodash.noop';

import { THEME_NAME, CONTEXT_MENU_SHOW, CONTEXT_MENU_HIDE } from '../constants';
import * as Styled from '../styled';
import VirtualReference from '../utils/VirtualReference';

const registerHandlers = handlers => {
  if (!Array.isArray(handlers)) {
    return;
  }

  for (const handler of handlers) {
    handler.target.addEventListener(
      handler.eventName,
      handler.callback,
      handler.capture,
    );
  }
};

const unregisterHandlers = handlers => {
  if (!Array.isArray(handlers)) {
    return;
  }

  for (const handler of handlers) {
    handler.target.removeEventListener(
      handler.eventName,
      handler.callback,
      handler.capture,
    );
  }
};

export class ContextMenu extends Component {
  static propTypes = {
    theme: PropTypes.object,
    children: PropTypes.node.isRequired,
    id: PropTypes.string.isRequired,
    testId: PropTypes.string,
    onShow: PropTypes.func,
    onHide: PropTypes.func,
    customCloseHandlers: PropTypes.array,
    onInsideClick: PropTypes.func,
  };
  static defaultProps = {
    testId: 'CONTEXT_MENU',
    onShow: noop,
    onHide: noop,
    customCloseHandlers: [],
    onInsideClick: noop,
  };
  constructor(props) {
    super(props);
    const { customCloseHandlers } = this.props;

    this.eventHandlers = [
      {
        target: document.body,
        eventName: 'click',
        callback: this.hide,
        capture: true,
      },
      {
        target: document.body,
        eventName: 'contextmenu',
        callback: this.hide,
        capture: true,
      },
      {
        target: window,
        eventName: 'resize',
        callback: this.hide,
        capture: false,
      },
      {
        target: window,
        eventName: 'scroll',
        callback: this.hide,
        capture: false,
      },
      ...customCloseHandlers,
    ];
  }
  state = {
    isVisible: false,
  };

  componentDidMount() {
    this.showToken = PubSub.subscribe(
      CONTEXT_MENU_SHOW,
      (event, { id: eventId, x, y }) => {
        const { id } = this.props;

        if (id !== eventId) {
          return;
        }
        this.show({ x, y });
      },
    );

    this.hideToken = PubSub.subscribe(
      CONTEXT_MENU_HIDE,
      (event, { id: eventId }) => {
        const { id } = this.props;

        if (id !== eventId) {
          return;
        }
        this.hide();
      },
    );
  }

  componentWillUnmount() {
    PubSub.unsubscribe(this.showToken);
    PubSub.unsubscribe(this.hideToken);
    unregisterHandlers(this.eventHandlers);

    this.virtualReference = undefined;
    this.showToken = undefined;
    this.hideToken = undefined;
  }

  show = ({ x, y }) => {
    const { onShow } = this.props;

    registerHandlers(this.eventHandlers);
    this.virtualReference = new VirtualReference(x, y);
    onShow();
    this.setState({ isVisible: true });
  };

  hide = () => {
    const { onHide } = this.props;
    const { isVisible } = this.state;

    unregisterHandlers(this.eventHandlers);

    if (isVisible) {
      this.wrapper(() => {
        onHide();
        this.setState({
          isVisible: false,
        });
      });
    }
  };

  virtualReference = undefined;
  showToken = undefined;
  hideToken = undefined;
  wrapper = (window.requestAnimationFrame || window.setTimeout).bind(window);

  render() {
    const {
      theme,
      children,
      onShow,
      onHide,
      onInsideClick,
      testId,
      ...otherProps
    } = this.props;
    const { isVisible } = this.state;

    if (!isVisible) {
      return null;
    }

    return (
      <Manager>
        <Popper
          referenceElement={this.virtualReference}
          placement="bottom-start"
          positionFixed
        >
          {({ ref, style }) => (
            <ThemeProvider theme={theme}>
              <Styled.ContextMenu
                onClick={onInsideClick}
                {...otherProps}
                ref={ref}
                style={style}
                {...getTestAttributes(testId)}
              >
                {children}
              </Styled.ContextMenu>
            </ThemeProvider>
          )}
        </Popper>
      </Manager>
    );
  }
}

export default withTheme(ContextMenu, THEME_NAME);
