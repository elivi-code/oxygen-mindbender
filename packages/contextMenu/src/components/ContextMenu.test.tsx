import React from 'react';
import 'jest-styled-components';
import { shallow, mount } from 'enzyme';
import { contextMenu as theme } from '@8x8/oxygen-constants';
import PubSub from 'pubsub-js';

import { ContextMenu } from './ContextMenu';
import { CONTEXT_MENU_SHOW, CONTEXT_MENU_HIDE } from '../constants';

jest.mock('../styled/ContextMenu', () => 'ContextMenu');

describe('<ContextMenu>', () => {
  let wrapper;
  const mockEventsMap = new Map();

  function setup(props = {}) {
    return shallow(
      <ContextMenu id="id" theme={theme} {...props}>
        <span>context menu</span>
      </ContextMenu>,
    );
  }

  function setupMount(props = {}) {
    return mount(
      <ContextMenu id="id" theme={theme} {...props}>
        <span>context menu</span>
      </ContextMenu>,
    );
  }

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render correctly', () => {
    wrapper = setup();
    wrapper.instance().show({ x: 0, y: 0 });
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render if not visible', () => {
    wrapper = setup();
    expect(wrapper.isEmptyRender()).toBe(true);
  });

  it('should subscribe to pubsub events after mounting', () => {
    const spy = jest.spyOn(PubSub, 'subscribe');

    wrapper = setupMount();

    expect(spy).toHaveBeenCalled();
  });

  it('should unsubscribe from pubsub events after unmounting', () => {
    const spy = jest.spyOn(PubSub, 'unsubscribe');

    wrapper = setupMount();
    wrapper.unmount();

    expect(spy).toHaveBeenCalled();
  });

  it('should call onShow when the context menu is shown', () => {
    const onShow = jest.fn();

    jest
      .spyOn(PubSub, 'subscribe')
      .mockImplementation((eventName, eventCallback) => {
        const currentEventCallbacks = mockEventsMap.get(eventName) || [];

        currentEventCallbacks.push(eventCallback);
        mockEventsMap.set(eventName, currentEventCallbacks);
      });

    jest
      .spyOn(PubSub, 'publish')
      .mockImplementation((eventName, eventParams) => {
        const currentEventCallbacks = mockEventsMap.get(eventName);

        currentEventCallbacks.forEach(cb => {
          cb(eventName, eventParams);
        });
      });

    wrapper = setupMount({ onShow });
    PubSub.publish(CONTEXT_MENU_SHOW, { id: 'id', x: 200, y: 200 });
    expect(onShow).toHaveBeenCalled();
  });

  it('should not call onShow when a different context menu is shown', () => {
    const onShow = jest.fn();

    jest
      .spyOn(PubSub, 'subscribe')
      .mockImplementation((eventName, eventCallback) => {
        const currentEventCallbacks = mockEventsMap.get(eventName) || [];

        currentEventCallbacks.push(eventCallback);
        mockEventsMap.set(eventName, currentEventCallbacks);
      });

    jest
      .spyOn(PubSub, 'publish')
      .mockImplementation((eventName, eventParams) => {
        const currentEventCallbacks = mockEventsMap.get(eventName);

        currentEventCallbacks.forEach(cb => {
          cb(eventName, eventParams);
        });
      });

    wrapper = setupMount({ onShow });
    PubSub.publish(CONTEXT_MENU_SHOW, { id: 'id2', x: 200, y: 200 });
    expect(onShow).toHaveBeenCalledTimes(0);
  });

  it('should call onHide when the context menu is closed', () => {
    const onHide = jest.fn();

    jest
      .spyOn(PubSub, 'subscribe')
      .mockImplementation((eventName, eventCallback) => {
        const currentEventCallbacks = mockEventsMap.get(eventName) || [];

        currentEventCallbacks.push(eventCallback);
        mockEventsMap.set(eventName, currentEventCallbacks);
      });

    jest
      .spyOn(PubSub, 'publish')
      .mockImplementation((eventName, eventParams) => {
        const currentEventCallbacks = mockEventsMap.get(eventName);

        currentEventCallbacks.forEach(cb => {
          cb(eventName, eventParams);
        });
      });

    jest.spyOn(window, 'requestAnimationFrame').mockImplementation(cb => cb());

    wrapper = setupMount({ onHide });
    PubSub.publish(CONTEXT_MENU_SHOW, { id: 'id', x: 200, y: 200 });
    PubSub.publish(CONTEXT_MENU_HIDE, { id: 'id' });
    expect(onHide).toHaveBeenCalled();
  });

  it('should not call onHide when a different context menu is closed', () => {
    const onHide = jest.fn();

    jest
      .spyOn(PubSub, 'subscribe')
      .mockImplementation((eventName, eventCallback) => {
        const currentEventCallbacks = mockEventsMap.get(eventName) || [];

        currentEventCallbacks.push(eventCallback);
        mockEventsMap.set(eventName, currentEventCallbacks);
      });

    jest
      .spyOn(PubSub, 'publish')
      .mockImplementation((eventName, eventParams) => {
        const currentEventCallbacks = mockEventsMap.get(eventName);

        currentEventCallbacks.forEach(cb => {
          cb(eventName, eventParams);
        });
      });

    wrapper = setupMount({ onHide });
    PubSub.publish(CONTEXT_MENU_SHOW, { id: 'id', x: 200, y: 200 });
    PubSub.publish(CONTEXT_MENU_HIDE, { id: 'id2' });
    expect(onHide).toHaveBeenCalledTimes(0);
  });

  it('should not call onHide if the context menu is not visible', () => {
    const onHide = jest.fn();

    jest
      .spyOn(PubSub, 'subscribe')
      .mockImplementation((eventName, eventCallback) => {
        const currentEventCallbacks = mockEventsMap.get(eventName) || [];

        currentEventCallbacks.push(eventCallback);
        mockEventsMap.set(eventName, currentEventCallbacks);
      });

    jest
      .spyOn(PubSub, 'publish')
      .mockImplementation((eventName, eventParams) => {
        const currentEventCallbacks = mockEventsMap.get(eventName);

        currentEventCallbacks.forEach(cb => {
          cb(eventName, eventParams);
        });
      });

    wrapper = setupMount({ onHide });
    PubSub.publish(CONTEXT_MENU_HIDE, { id: 'id' });
    expect(onHide).toHaveBeenCalledTimes(0);
  });
});
