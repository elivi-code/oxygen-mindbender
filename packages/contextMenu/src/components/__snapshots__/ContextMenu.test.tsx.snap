// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContextMenu> should render correctly 1`] = `
<Manager>
  <Popper
    placement="bottom-start"
    positionFixed={true}
    referenceElement={
      VirtualReference {
        "clientRect": {
          "bottom": 0,
          "height": 0,
          "left": 0,
          "right": 0,
          "top": 0,
          "width": 0,
        },
      }
    }
  >
    <Component />
  </Popper>
</Manager>
`;
